ClinGen/clincoded
# Connects to #1341. Connects to #1342. Allowing multiple curators to add evidence.

**Known issues:**
1. The **Last edited** in the **CuratorRecordHeader** does not currently reflect the very last user who _add_ a new object to a parent object (e.g. _a new score object_ to _a parent evidence object_). This is because we generally save the new object first, and then we update the parent object with the returned **UUID** of the new object. And so the parent object would have _most recent_ **last_modified** timestamp. A new ticket (#1344) has been created for R12 to address this issue.

**Steps to test #1341:**
1. In GCI, go into a GDM record of which you are not the creator.
2. Also select a PMID that was not added to this GDM by you.
3. In the Curation Palette on the right, expect to see that you are allowed to add evidence to either one of the 5 evidence categories (_Group, Family, Individual, Case-Control and Experimental Data_).
4. Expect to be able to edit your own evidence.
![image](https://cloud.githubusercontent.com/assets/6956310/26321774/12a99a04-3edd-11e7-9160-f42a2bc1ee7f.png)
5. Expect to be able to view/score evidence added by the others, but not able to edit them.
6. In the **CuratorRecordHeader**, expect to see your name being displayed as one of the **Participants** even though you have not added a new PMID.
![image](https://cloud.githubusercontent.com/assets/6956310/26321756/fd2fb4e2-3edc-11e7-978d-d02ae2f394d2.png)
7. In the **CuratorRecordHeader**, expect to see **Last edited** displays the timestamp for the record updated last.
8. Now go to the GDM list page (e.g. /gdm), expect to see your name being displayed in the **Participants** column of the GDM in question, as well as the timestamp for the record updated last in **Last Edited** column.
![image](https://cloud.githubusercontent.com/assets/6956310/26321799/2867fc96-3edd-11e7-8023-8cd0a4f079a2.png)

**Steps to test #1342:**
1. In GCI, go into a GDM record of which you are not the creator.
2. Also select a PMID that was not added to this GDM by you.
3. In the Curation Palette on the right, select **VIew/Score** on the evidence owned by others but not yet been scored by the evidence creator.
![image](https://cloud.githubusercontent.com/assets/6956310/26321774/12a99a04-3edd-11e7-9160-f42a2bc1ee7f.png)
4. On the view-only page of the evidence, expect to see a message in the yellow warning box in the very last panel on the page.
![image](https://cloud.githubusercontent.com/assets/6956310/26321955/d16286ae-3edd-11e7-9036-473820fb4ae0.png)
